tt_metal/impl/dispatch/command_queue.hpp

```diff
@@ -93,6 +97,7 @@ class EnqueueReadBufferCommand : public Command {
         std::optional<uint32_t> pages_to_read = std::nullopt);
 
     const DeviceCommand assemble_device_command(uint32_t dst);
+    const DeviceCommand assemble_device_command_sharded(uint32_t dst);
```

Comment:
If you do what Bill suggested, we won't need an assemble_device_command_sharded. It'll just be the assemble_device_command for EnqueueWrite/ReadBuffer